src/utils/PathUtils.ts

```diff
@@ -106,13 +107,98 @@ function convertSoftIndexes(element: PathPart, pathMap: Map<string, number>) {
   }
 }
 
+/**
+ *
```

Comment:
Since there is a strict and not strict version of this function now, it might be good to provide a brief description so it's clear what the difference is between this version and the not strict version.